src/autowiring/DispatchQueue.cpp

```diff
@@ -87,12 +54,82 @@ void DispatchQueue::DispatchEventUnsafe(std::unique_lock<std::mutex>& lk) {
   (*thunk)();
 }
 
+void DispatchQueue::Abort(void) {
+  std::lock_guard<std::mutex> lk(m_dispatchLock);
+  m_aborted = true;
+  
+  // Do not permit any more lambdas to be pended to our queue:
+  m_dispatchCap = 0;
+  
+  // Destroy the whole dispatch queue:
+  while(!m_dispatchQueue.empty()) {
+    delete m_dispatchQueue.front();
+    m_dispatchQueue.pop_front();
+  }
+  
+  // Wake up anyone who is still waiting:
+  m_queueUpdated.notify_all();
+}
+
 bool DispatchQueue::DispatchEvent(void) {
   std::unique_lock<std::mutex> lk(m_dispatchLock);
-  if(m_dispatchQueue.empty())
+
+  // Update queue with delayed events that are ready
+  PromoteReadyEventsUnsafe();
+
+  if (m_dispatchQueue.empty())
     return false;
 
   DispatchEventUnsafe(lk);
   return true;
 }
 
+int DispatchQueue::DispatchAllEvents(void) {
+  int retVal = 0;
+  while(DispatchEvent())
+    retVal++;
+  return retVal;
+}
+
+void DispatchQueue::AddExisting(DispatchThunkBase* pBase) {
+  std::unique_lock<std::mutex> lk(m_dispatchLock);
+  if(m_dispatchQueue.size() >= m_dispatchCap)
+    return;
+  
+  m_dispatchQueue.push_back(pBase);
+  m_queueUpdated.notify_all();
+  OnPended(std::move(lk));
+}
+
+std::chrono::steady_clock::time_point
+DispatchQueue::SuggestSoonestWakeupTimeUnsafe(std::chrono::steady_clock::time_point latestTime) const {
+  return
+  m_delayedQueue.empty() ?
```

Comment:
Add two more spaces here and in the lines following--these are part of the `return` statement starting on line 104.